packages/instrumentation-pinecone/test/instrumentation.test.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Traceloop
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { context } from "@opentelemetry/api";
+import { AsyncHooksContextManager } from "@opentelemetry/context-async-hooks";
+import { PineconeInstrumentation } from "../src/instrumentation";
+import * as assert from "assert";
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from "@opentelemetry/sdk-trace-base";
+import { Pinecone, Index } from "@pinecone-database/pinecone";
+import * as pc_module from "@pinecone-database/pinecone";
+
+const memoryExporter = new InMemorySpanExporter();
+const pc = new Pinecone();
+
+describe("Test LlamaIndex instrumentation", () => {
+  const provider = new BasicTracerProvider();
+  let instrumentation: PineconeInstrumentation;
+  let contextManager: AsyncHooksContextManager;
+  let pc_index: Index;
+
+  before(async () => {
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    instrumentation = new PineconeInstrumentation();
+    instrumentation.setTracerProvider(provider);
+    instrumentation.manuallyInstrument(pc_module);
```

Comment:
@nirga I'm not sure why but I need to call `manuallyInstrument` because otherwise the wrappers are not working. Could you help me debug this?

To replicate this, you can comment out line 42 here and try running the tests.

Reply:
@kartikay-bagla you need to initialize the instrumentation in the SDK, [here](https://github.com/traceloop/openllmetry-js/blob/eb364d985e1aeb1d6067f97a520b277f0f797f34/packages/traceloop-sdk/src/lib/tracing/index.ts#L34)

Reply:
Already did that, check [here](https://github.com/traceloop/openllmetry-js/blob/42165f1b0261c2eda906e945c56394eb3bf1938d/packages/traceloop-sdk/src/lib/tracing/index.ts#L35) and [here](https://github.com/traceloop/openllmetry-js/blob/42165f1b0261c2eda906e945c56394eb3bf1938d/packages/traceloop-sdk/src/lib/tracing/index.ts#L116).

Reply:
And it still doesn't work for you without manually instrument?

Reply:
yep